### PR TITLE
Edit Dashboard title and Slice title in place

### LINKED ISSUE #2796: Edit Dashboard title and Slice title in place
Currently the only way to edit the title of a dashboard or a slice is to use the CRUD view provided by FAB. 

Instead of that, it'd be nice to have a way to edit the name in-place, the same way that it's done in Google docs for instance.

### PR DESCRIPTION
**Pull request by graceguo-supercat**
Add EditableTitle component into Dashboard and Explore view to support edit title inline.

fixes https://github.com/airbnb/superset/issues/2796

dashboard view:
zi7a4nbzei

Explore view:
slice

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -503,7 +533,8 @@ def test_dashboard_with_created_by_can_be_accessed_by_public_users(self):
         )
         self.grant_public_access_to_table(table)
 
-        dash = db.session.query(models.Dashboard).filter_by(dashboard_title="Births").first()
+        dash = db.session.query(models.Dashboard).filter_by(
```
Comment (mistercrunch) [inline tests/core_tests.py]: nit: I've been breaking long sqlalchemy queries method-chaining-style, but in python it needs parenthesis as in:
```
dash = (
    db.session.query(models.Dashboard)
    .filter_by(slug="births")
    .first()
)
```

File: superset/assets/javascripts/explore/components/ChartContainer.jsx [hunk 2]
```diff
@@ -145,6 +150,19 @@ class ChartContainer extends React.PureComponent {
     this.props.actions.runQuery(this.props.formData, true);
   }
 
+  updateChartTitle(newTitle) {
+    const params = {};
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/components/ChartContainer.jsx]: nit: this is a bit more elegant:
```
const params = {
    slice_name: newTitle,
    action: 'overwrite',
    form_data = this.props.formData,
};
```

File: superset/assets/javascripts/explore/components/ChartContainer.jsx [hunk 3]
```diff
@@ -145,6 +150,19 @@ class ChartContainer extends React.PureComponent {
...
+    params.slice_name = newTitle;
+    params.action = 'overwrite';
+    params.form_data = this.props.formData;
+    const saveUrl = '/superset/explore/' +
+      `${this.props.datasourceType}/${this.props.datasourceId}/?${serialize(params)}`;
```
Comment (mistercrunch) [inline superset/assets/javascripts/explore/components/ChartContainer.jsx]: Maybe we can reuse the getExploreUrl method here? https://github.com/airbnb/superset/blob/master/superset/assets/javascripts/explore/exploreUtils.js#L4

Or maybe it's time soon to remove move save/overwrite out of `/superset/explore/` and into its own POST endpoint.

File: superset/assets/utils/common.js [hunk 4]
```diff
@@ -86,3 +86,33 @@ export function getShortUrl(longUrl, callback) {
     },
   });
 }
+
+export function serialize(obj) {
```
Comment (mistercrunch) [inline superset/assets/utils/common.js]: I'm hoping we can get away without this function by reusing `getExploreUrl`

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Really excited to see this feature!

File: tests/core_tests.py (hunk 1 above)
Comment (graceguo-supercat) [inline tests/core_tests.py]: fixed

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**